nemo/collections/common/data/dataset.py

```diff
@@ -286,3 +289,313 @@ def __len__(self):
     def __getitem__(self, idx):
         dataset_id, dataset_index = self.indices[idx]
         return self.datasets[dataset_id][dataset_index]
+
+
+class CodeSwitchedDataset(IterableDataset):
```

Comment:
Seems the dataset dosnt support tarred dataset. Most of our datasets are too vast to use loose file anymore.. Maybe refactor out the core components and wrap them to support at least tarred datasets (bucketing is also important but can be done later)

Reply:
Or is it a wrapper dataset like bucketing already? Im a bit confused so ill leave it to @bmwshop to review properly

Reply:
Probably needs to be renamed a little if its a wrapped dataset

Reply:
Correct, it's a wrapped dataset, just like ConcatDataset which I used as the base for this new class

Reply:
Right, it is a wrapper class.